enhancements/update/change-management-and-maintenance-schedules.md

```diff
@@ -0,0 +1,1273 @@
+to iteratively find permissive windows at the cluster / control-plane level. These can be added to an 
+[interval-tree](https://github.com/rdleal/intervalst) . As dates are added, rrule calculations for the worker-nodes
+can be performed. The interval-tree should be able to efficiently determine whether there is an
+intersection between the permissive intervals it has stored for the control-plane and the time range tested for the 
+worker-nodes.
+
+Since it is possible there is no overlap, limits must be placed on this search. Once dates >1000 days from
+the present moment are being tested, the operator can behave as if the next window will occur in
+1000 days (prevents infinite search for overlap).
+
+This outcome does not need to be recomputed unless the operator restarts Or one of the RRULE involved
+is modified.
+
+If an overlap _is_ found, no additional intervals need to be added to the tree and it can be discarded.
+The operator can store the start & end datetimes for the overlap and count down the seconds remaining 
+until it occurs. Obviously, this calculation must be repeated:
+1. If either MaintenanceSchedule configuration is updated.
+1. The operator is restarted.
+1. At the end of a permissive window, in order to determine the next permissive window.
+
+
+#### Service Delivery Option Sanitization
+It is obvious that the range of flexible options provided by change management configurations offers
+can create risks for inexperienced cluster lifecycle administrators. For example, setting a 
+standalone cluster to use the Assisted strategy and failing to trigger worker-node updates will
+leave unpatched CVEs on worker-nodes much longer than necessary. It will also eventually lead to
+the need to resolve version skew (Upgradeable=False will be reported by the API cluster operator).
+
+Service Delivery understands that exposing the full range of options to cluster
+lifecycle administrators could dramatically increase the overhead of managing their fleet. To
+prevent this outcome, Service Delivery will only expose a subset of the change management
+strategies. They will also implement sanitization of the configuration options a user can
+supply to those strategies. For example, a simplified interface in OCM for building a
+limited range of RRULEs that are compliant with Service Delivery's update policies.
+
+#### Node Disruption Policy
+https://github.com/openshift/enhancements/pull/1525 describes the addition of `nodeDisruptionPolicy`
+to `MachineConfiguration`. Through this configuration, an administrator can convey that
+a configuration should not trigger a node to be rebooted.
+
+When `nodeDisruptionPolicy` indicates that a `MachineConfiguration` should not trigger
+a node reboot, it becomes a non-material change from the perspective of a maintenance
+schedule. In other words, it can be applied immediately, even outside a permissive window.

```

Comment:
I don't think that's safe, since disruption is not just reboots, but also node drains, etc. all of which you can specify via the policy. Besides, the MCO isn't that granular that it can break down a subset of MC changes to happen.

I think node disruption policy should only define how a change is acted on, but otherwise is gated by the maint. schedule like everyone else.